pkg/console/operator/sync_v400.go

```diff
@@ -329,12 +331,18 @@ func (co *consoleOperator) SyncConfigMap(
 		}
 	}
 
+	pluginsList, pluginsListErr := co.consolePluginClient.List(ctx, metav1.ListOptions{})
+	if pluginsListErr != nil {
+		return nil, false, "FailedGetConsolePlugins", pluginsListErr
+	}
+	enabledPlugins := getEnabledPlugins(operatorConfig, pluginsList.Items)
```

Comment:
:+1: This means the ConfigMap will contain an intersection of enabled plugins (operator config) and available plugins (`ConsolePlugin` custom resources).

@spadgett This shouldn't interfere with enabling a dynamic plugin as part of operator install flow, right? Console UI can start loading a plugin only when its assets are deployed and available via Service, which means only after the operator install process completes.

@jhadvig Will the ConfigMap be updated upon a change of either operator config or `ConsolePlugin` custom resources?

So if I understand correctly, operator config tracks all plugins marked as enabled (covers operator install flow and plugin admin page) and the ConfigMap tracks enabled plugins which should be loaded in the UI.

Reply:
@vojtechszocs 
> This shouldn't interfere with enabling a dynamic plugin as part of operator install flow, right?

No it shouldn't. It's an mechanism that prevents console from trying to load plugins that are not present on the cluster as a `ConsolePlugin` CR

> Will the ConfigMap be updated upon a change of either operator config or ConsolePlugin custom resources?

Yes. Both changes should redeploy the console pods

>  So if I understand correctly, operator config tracks all plugins marked as enabled (covers operator install flow and plugin admin page) and the ConfigMap tracks enabled plugins which should be loaded in the UI.

Yes. operator config tracks all the enabled plugins, but since we can not guarantee that are of these plugins are available on the cluster (since the operator config can be updated manually as well) operator will check if a corresponding `ConsolePlugin` exists on the cluster. If so then it will put int into the `console-config.yaml`.